rrule/src/core/utils.rs

```diff
@@ -61,14 +61,18 @@ where
         }
     }
     // Make sure that the user always know when there are more dates.
-    if list.len() > u16::MAX as usize {
-        (
-            list,
-            Some(RRuleError::new_iter_err(format!(
-                "List reached maximum limit (`{}`), so there might be more items.",
-                u16::MAX
-            ))),
-        )
+    if let Some(limit) = limit {
+        if usize::from(limit) < list.len() {
+            (
+                list,
+                Some(RRuleError::new_iter_err(format!(
+                    "List reached maximum limit (`{}`), so there might be more items.",
```

Comment:
A question, can we guesstimate the number of iterations, even before iterating? With a formula like any of these:
- `COUNT`
- (`min(.to_date(), UNTIL)` - `DTSTART`) / `FREQ`
- ...

Reply:
It is an interesting observation. I think it would be possible to set an upper bound on RRule, if count or until is defined. If neither count nor until is defined, then that limit is infinite and we will encounter an infinite loop if we haven't set a limit.